core/storage/user/gae_models.py

```diff
@@ -1475,6 +1483,18 @@ class ExplorationUserDataModel(base_models.BaseModel):
     # The user's preference for receiving feedback emails for this exploration.
     mute_feedback_notifications = datastore_services.BooleanProperty(
         default=feconf.DEFAULT_FEEDBACK_NOTIFICATIONS_MUTED_PREFERENCE)
+    # The state name of the furthest reached checkpoint.
+    furthest_reached_checkpoint_state_name = datastore_services.StringProperty(
+        default=None)
+    # The exploration version of the furthest reached checkpoint.
+    furthest_reached_checkpoint_exp_version = (
+        datastore_services.IntegerProperty(default=None))
+    # The state name of the most recently reached checkpoint.
+    most_recently_reached_checkpoint_state_name = (
+        datastore_services.StringProperty(default=None))
+    # The exploration version of the most recently reached checkpoint.
+    most_recently_reached_checkpoint_exp_version = (
+        datastore_services.IntegerProperty(default=None))
```

Comment:
Do we want to mention that it is the **exp** version, wouldn't it be enough to just mention version?

Reply:
I think having **exp** version in the variable name would suggest clearly what this variable is actually referring to (exploration version of the most recently reached checkpoint). A variable named `most_recently_reached_checkpoint_version` sounds more like as if checkpoints have their own versions. Thoughts?

Reply:
Ah, yes, you're right.